.github/actions/environment/action.yml

```diff
@@ -35,15 +35,19 @@ runs:
       run: dotnet new globaljson --sdk-version 7.0.203 --force
       shell: bash
 
-    - name: Dependency Caching
+    # We don't use a lockfile, so hash all files where we might be keeping <PackageReference> tags.
+    - name: Collect the list of NuGet package references for caching
+      shell: bash
+      run: |
+        grep -hr "<PackageReference " --include="*proj" --include="*.props" . | sort -u  > package-references.txt
```

Comment:
Should we include `*.targets` files in this hash as well?

Reply:
the relavant code has been removed so resolving this.